cpp/src/rolling/rolling.cu

```diff
@@ -131,8 +135,16 @@ void gpu_rolling(gdf_size_type nrows,
     gdf_index_type const out_mask_location = cudf::util::detail::bit_container_index<bit_mask_t, gdf_index_type>(i);
 
     // only one thread writes the mask
-    if (0 == threadIdx.x % warpSize)
+    if (0 == threadIdx.x % warpSize) {
       out_col_valid[out_mask_location] = result_mask;
+      // Perform null bitmask null count
+      valid_count_accumulate += cudf::detail::single_lane_popc_block_reduce(result_mask);
```

Comment:
There's no way this works. The function you are calling calls __syncthreads(), which means you can't call it inside a divergent branch...

Reply:
This is my first kernel program :slightly_smiling_face:. I had verified the null_count in Java layer and it was working fine :stuck_out_tongue: .  Maybe because the dataset was so small that block/grid didn't come into consideration and it passed by chance?
```
@Test
  void testWindowStatic1() {
    WindowOptions v0 = WindowOptions.builder().windowSize(2).minPeriods(2).forwardWindow(1).
        aggType(AggregateOp.SUM).build();
    try (ColumnVector v1 = ColumnVector.fromBoxedInts(5, null, 7, null, 8);
         ColumnVector expected = ColumnVector.fromBoxedInts(null, 12, null, 15, null);
         ColumnVector result = v1.rollingWindow(v0)) {
      result.ensureOnHost();
      assertTrue(result.hasNulls());
      assertColumnsAreEqual(result, expected);
    }
  }
````

Not able to wrap my head around this below problem. 
If I move the  computation of `single_lane_popc_block_reduce`  outside of `if` condition, then `valid_count_accumulate` is set to  `1260`(instead of number of valid bits) and hence getting wrong results.  

This is only change I have made:
```
 // only one thread writes the mask
    if (0 == threadIdx.x % warpSize) {
      out_col_valid[out_mask_location] = result_mask;
    }
    // Perform null bitmask null count
    valid_count_accumulate += cudf::detail::single_lane_popc_block_reduce(result_mask);

    if (threadIdx.x == 0) {
      atomicAdd(output_null_count, valid_count_accumulate);
    }
```